src/rabbit_delayed_message.erl

```diff
@@ -161,11 +161,18 @@ maybe_delay_first() ->
     end.
 
 route(#delay_key{exchange = Ex}, Deliveries) ->
+    ExName = Ex#exchange.name,
     lists:map(fun (#delay_entry{delivery = D}) ->
                       D2 = swap_delay_header(D),
                       Dests = rabbit_exchange:route(Ex, D2),
                       Qs = rabbit_amqqueue:lookup(Dests),
-                      rabbit_amqqueue:deliver(Qs, D2)
+                      case rabbit_queue_type:deliver(Qs, D2, stateless) of
```

Comment:
I'm not sure if it makes sense to check the result. If delivery to the first queue succeeds but fails for the second queue, the metrics will be incorrect anyway. I'm also not familiar with what stateless vs non-stateless delivery means.

Reply:
This would be good enough for most users. Accurate metrics for events that happen in the future and reference things that might no longer exist (e.g. for error communication) is not something that most would expect.